weave-js/src/components/RunLink.tsx

```diff
@@ -0,0 +1,217 @@
+
+type TagColor = {
+  id: string;
+  name: string;
+  colorIndex: number;
+};
+
+type RunData = {
+  name: string;
+  displayName: string;
+  createdAt: string;
+  notes: string;
+  state: string;
+  tagColors: TagColor[];
+};
+
+const FETCH_PROJECT_RUN_QUERY = gql`
+  query FetchProjectRun(
+    $entityName: String!
+    $projectName: String!
+    $runName: String!
+  ) {
+    project(entityName: $entityName, name: $projectName) {
+      id
+      internalId
+      run(name: $runName) {
+        id
+        name
+        displayName
+        createdAt
+        notes
+        state
+        tagColors {
+          id
+          name
+          colorIndex
+        }
+      }
+    }
+  }
+`;
+
+type RunResult = 'load' | 'loading' | 'error' | RunData;
+
+const RunContentHeader = styled.div`
+  display: flex;
+  align-items: center;
+  padding: 2px 2px 2px 8px;
+  background-color: ${Colors.MOON_100};
+`;
+RunContentHeader.displayName = 'S.RunContentHeader';
+
+const RunDisplayName = styled.div`
+  font-weight: 600;
+  flex: 1 1 auto;
+`;
+RunDisplayName.displayName = 'S.RunDisplayName';
+
+const RunContentBody = styled.div`
+  padding: 4px;
+  display: flex;
+  align-items: flex-start;
+  gap: 4px;
+`;
+RunContentBody.displayName = 'S.RunContentBody';
+
+type RunContentProps = {run: RunData};
+
+const RunContent = ({run}: RunContentProps) => {
+  return (
+    <Tailwind>
+      <DraggableHandle>
+        <RunContentHeader>
+          <RunDisplayName>{run.displayName}</RunDisplayName>
+        </RunContentHeader>
+      </DraggableHandle>
+      <RunContentBody>
+        <div className="grid grid-cols-[auto_auto] gap-4">
+          {run.notes && (
+            <>
+              <div className="text-right font-semibold">Notes</div>
+              <div>{run.notes}</div>
+            </>
+          )}
+          {run.tagColors.length > 0 && (
+            <>
+              <div className="text-right font-semibold">Tags</div>
+              <div className="flex items-center gap-4">
+                {run.tagColors.map(tag => (
+                  <Tag
+                    key={tag.id}
+                    label={tag.name}
+                    color={getTagContrastColor(tag.colorIndex)}
+                  />
+                ))}
+              </div>
+            </>
+          )}
+          <div className="text-right font-semibold">State</div>
+          <div>
+            <RunState value={run.state} />
+          </div>
+          <div className="text-right font-semibold">Start time</div>
+          <div>
+            <Timestamp value={run.createdAt} />
+          </div>
+        </div>
+      </RunContentBody>
+    </Tailwind>
+  );
+};
+
+export const useProjectRun = (
+  entityName: string,
+  projectName: string,
+  runName: string
+) => {
+  const apolloClient = useApolloClient();
+
+  const [runData, setRunData] = useState<RunResult>('load');
+  useEffect(() => {
+    let mounted = true;
+    setRunData('loading');
+    apolloClient
+      .query({
```

Comment:
More N+1 queries 😢 
Can we query runs in bulk once per page?

Reply:
Agree it is 😢 but N+1 is how it already works, not something changed in this PR. Bulk querying runs information would be a more complicated change that I don't think we should block this improvement on. We do also have client-side caching so it's not quite as bad as it seems, especially since a common case will be only viewing the calls for the selected run.